### ISSUE TITLE
file URI schema dereferencing messes up in DISLoader

### ISSUE THREAD

**Issue by daoane** [corewall/corelyzer#8]
Corelyzer 2.10 on Ubuntu Linux 20.04 LTS
(the spelling 'Corelizer' (below) is a mistake in the new mDIS exporter and, thus, correct in the context and not the cause for error)

I try to import core scans that were exported from the mDIS. XML file and images are in the same directory. The XML file contains the correct full path to the image of each core section. I open the XML file by File => DIS => Import

In the terminal I see the following
```
---> [DISLoader] Read DIS file: '/home/hl/Desktop/CorelizerXmlExport_COSC-2B/ICDP_5054_2_B.xml' with DIS-Prefix: ''
EXP> name: Collisional Orogeny in the Scandinavian Caledonides, id: 1
SITE> name: COSC-2, id: 2
HOLE> name: B, id: 5
CORE> name: 1Z, id: 2174, topDepth: 6.08, mcd_depth: 6.08, length: 2.52
SECTION> name: 1, id: 6124, topDepth: 6.08, mcdDepth: 6.08, length: 0.91
IMAGE> name: CS_5054_2_B_1_1.jpg, id: 15204, url: file:/home/hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg
imageUrl is: file:/home/hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg
[URL OK] the path is: /home/hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg
[URL OK] the file is: /home/hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg
[DISLoader] !Mac's imageUrl: file://///hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg
```
Obviously, the URL is interpreted correctly until the second line from below. When the DISLoader comes in, something strange happens - too me it looks like a functionality that is intended to work on a Mac, although I can't see the difference to Linux. A full path is a full path. In any case, it results in the Corelyzer error "File could not be found: '/hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg'"

**Comment by daoane** [comment]
One more test - all three variants should work according to file URI specs. 

`url="file:/home/hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg"` results in "File could not be found: '/hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg'"

`file:///home/hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg"` results in "File could not be found: '/me/hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg'"

`url="file://localhost/home/hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg"` results in "File could not be found: '/alhost/home/hl/Desktop/CorelizerXmlExport_COSC-2B/CS_5054_2_B_1_1.jpg'"

The bits that are missing are "/home", "///ho" and "//loc", respectively. So obviously "something" is cutting off five characters from the correct path and thereby causes the problem. I'll see whether I can find the place in the code.

**Comment by daoane** [comment]
This is the culprit, but I don't exactly understand how characters are cut and, more importantly, why this is invoked at all with Linux.

https://github.com/corewall/corelyzer/blob/411f90f18ffcfe588ef819f2ce99622050e0cfd3/app/src/main/java/corelyzer/io/DISLoader.java#L468

**Comment by sorghumking** [comment]
Fixed by 5786a158a3d0f34edbf87957aae1a3cf54649eae
